libs-haskell/da-hs-base/src/Data/Aeson/TH/Extended.hs

```diff
@@ -42,27 +40,24 @@ daAesonEncodingOptions prefix = defaultOptions
         (x:xs) -> toLower x : xs
         []     -> []
 
-deriveDAJSON :: String -> TH.Name -> TH.Q [TH.Dec]
-deriveDAJSON prefix = deriveJSON $ daAesonEncodingOptions prefix
-
 deriveDAToJSON :: String -> TH.Name -> TH.Q [TH.Dec]
 deriveDAToJSON = deriveToJSON . daAesonEncodingOptions
 
 deriveDAFromJSON :: String -> TH.Name -> TH.Q [TH.Dec]
 deriveDAFromJSON = deriveFromJSON . daAesonEncodingOptions
 
--- | Like 'daAesonEncodingOptions', but will strip and make lowercase
--- also constructor tags, on top of field modifiers.
-daAesonEncodingOptionsTagged :: String -> Options
-daAesonEncodingOptionsTagged prefix = (daAesonEncodingOptions prefix)
-    { constructorTagModifier = \label ->
-        maybe label lowerHead $ stripPrefix prefix label
-    }
-  where
-    lowerHead = \case
-        (x:xs) -> toLower x : xs
-        []     -> []
 
--- | Like 'deriveDAToJSON', but uses 'daAesonEncodingOptionsTagged'
-deriveDAToJSONTagged :: String -> TH.Name -> TH.Q [TH.Dec]
-deriveDAToJSONTagged = deriveToJSON . daAesonEncodingOptionsTagged
+-- | Generate a lens for every field in a record. The name of the lens is the
+-- name of the field prefixed by an underscore. For instance, for
+--
+-- > data Foo = Foo{bar :: Int, _baz :: Bool}
+--
+-- it will generate
+--
+-- > _bar :: Lens' Foo Int
+-- > __baz :: Lens' Foo Bool
+makeUnderscoreLenses :: TH.Name -> TH.DecsQ
```

Comment:
I'm not convinced this module is the right place for this function. I think we can remove this module once we use`haskell-lsp`. Why not leave this function where it was before, that is, inlined at its only use site?

Reply:
OK, I dropped this change from the patch. Still feels odd where it is, but let's deal with that another time.